File: docs/Algorithms/Sort/insertion_sort.py

def insertion_sort(arr):
    n = len(arr)
    for i in range(1, n):
        curr_val = arr[i]
        idx = i - 1

        while idx >= 0 and curr_val < arr[idx]:
            arr[idx+1] = arr[idx]
            idx -= 1
        arr[idx+1] = curr_val 
    
    return arr


sorted_arr = insertion_sort([ 7, 12, 9, 11, 3])
print(sorted_arr)

File: docs/Algorithms/Sort/insertion_sort.cpp
#include <iostream>

using namespace std;

void insertion_sort(int arr[], int n){
    for (int i = 1; i < n; i++) {
        int curr_val = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > curr_val) {
            arr[j+1] = arr[j];
            --j;
        };
        arr[j+1] = curr_val; 
    };
    for (int i=0; i<n; i++) {
        cout << arr[i] << " ";
    };
};


int main() {
    int arr[] = {7, 12, 9, 11, 3};
    int n = sizeof(arr) / sizeof(arr[0]);
    insertion_sort(arr, n);
    return 0;
};